src/server/pps/server/api_server.go

```diff
@@ -1344,6 +1350,56 @@ func (a *apiServer) QueryLoki(request *pps.LokiRequest, apiQueryLokiServer pps.A
 	})
 }
 
+func (a *apiServer) CheckStatus(request *pps.CheckStatusRequest, apiCheckStatusServer pps.API_CheckStatusServer) (retErr error) {
+	ctx := apiCheckStatusServer.Context()
+	checkProjectAccess := miscutil.CacheFunc(func(project string) error {
+		return a.env.AuthServer.CheckProjectIsAuthorized(ctx, &pfs.Project{Name: project}, auth.Permission_PROJECT_LIST_REPO)
+	}, 100)
+	checkAccess := func(ctx context.Context, pipeline *pps.Pipeline) error {
+		if err := checkProjectAccess(pipeline.Project.GetName()); err != nil {
+			if !errors.As(err, &auth.ErrNotAuthorized{}) {
+				return err
+			}
+			return a.env.AuthServer.CheckRepoIsAuthorized(ctx, &pfs.Repo{Name: pipeline.Name, Type: pfs.UserRepoType, Project: pipeline.Project}, auth.Permission_REPO_READ)
+		}
+		return nil
+	}
+	project := []*pfs.Project{}
+	if request.GetProject().GetName() != "" {
+		project = append(project, request.GetProject())
+	}
+	filterPipeline, err := newMessageFilterFunc("", project)
+	if err != nil {
+		return errors.Wrap(err, "error creating message filter function")
+	}
+	var alerts []string
+	if err := ppsutil.ListPipelineInfo(ctx, a.pipelines, nil, 0, func(pipelineInfo *pps.PipelineInfo) error {
+		if ok, err := filterPipeline(ctx, pipelineInfo); err != nil {
+			return errors.Wrap(err, "error filtering pipeline")
+		} else if !ok {
+			return nil
+		}
+		zero := &timestamppb.Timestamp{}
+		if checkAccess(ctx, pipelineInfo.Pipeline) == nil &&
```

Comment:
I think these access checks are always a source of confusion; you should log something when access is not granted and so the pipeline is not included in the results.